samples/financial_transcripts/generate_benchmark_qa.ipynb

```diff
@@ -0,0 +1,562 @@
+    "        Answer: example answer goes here\n",
+    "\n",
+    "        Based on ticker, quarter, year, the question can be phrased in different ways e.g. MSFT FY23 Q1, MSFT FY2023 1st quarter, e.t.c.\n",
+    "        In case the text question is not relevant, please skip the question and answer pair.\n",
+    "        input_text1: \n",
+    "        {chunk_text1}\n",
+    "        input_text2:\n",
+    "        {chunk_text2}\n",
+    "        ticker: {ticker}\n",
+    "        quarter: {quarter}\n",
+    "        year: {year}\n",
+    "        \"\"\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Randomly pick filter parameters:\n",
+    "\n",
+    "Add Filter Parameters to randomly pick them for extracting context (chunks). This will help diversify generating questions and answers. \n",
+    "In the MSFT Financial Transcripts use-case, the Ticker name is MSFT, but you can easily add other ticker labels for a larger Financial dataset. `Year`,`Quarter`, and `Id` are the key parameters used in this use-case. In this notebook, we are using two random chunks (chunk ids) from specified filter keys (year, quarter).\n",
+    "\n",
+    "###### TODO: Add Tools (an updated version of function calls) once it is available for working with newer models. Function Calls are currently deprecated for gpt models with versions beyond 07-01-2023."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Ticker = np.random.choice(df['Ticker'].unique())\n",
+    "Year = np.random.choice(df[df['Ticker']==Ticker]['Year'].unique())\n",
+    "Quarter = np.random.choice(df[(df['Ticker']==Ticker) & (df['Year']==Year)]['Quarter'].unique())\n",
+    "Id = np.random.choice(df[(df['Quarter']==Quarter) & (df['Year']==Year)]['Id'].unique())\n",
+    "Id2 = np.random.choice(df[(df['Quarter']==Quarter) & (df['Year']==Year) & (df['Id']!=Id)]['Id'].unique())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'MSFT'"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "Ticker"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Generate Questions (using Azure OpenAI only)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "1. What was the percentage increase in server products and cloud services revenue for MSFT in Q1 FY23?\n",
+      "Answer: The server products and cloud services revenue increased by 22% in constant currency for MSFT in Q1 FY23.\n",
+      "\n",
+      "2. How much did Azure and other cloud services revenue grow in constant currency for MSFT in Q1 FY23?\n",
```

Comment:
Same issue as with q1, 'Azure and other cloud services revenue grew 35 percent and 42 percent in constant currency' - the 42 percent part referring to 'other cloud services' was ignored.